ansible/roles/install_prerequisites/tasks/main.yaml

```diff
@@ -0,0 +1,15 @@
+- name: Install git & python3
+  yum:
+    name: "{{ item }}"
+    state: present
+    update_cache: yes
+  loop:
+    - git
+    - python3
+
+- name: Install poetry
+  shell:
+    cmd: curl -sSL https://install.python-poetry.org | python3 -
+    creates: /usr/local/bin/poetry
+  register: install_poetry
+  changed_when: install_poetry.changed
```

Comment:
Yes it does:
```
ok: [worker_node] => {
    "msg": {
        "ansible_facts": {
            "discovered_interpreter_python": "/usr/local/bin/python3.9"
        },
        "changed": false,
        "cmd": "curl -sSL https://install.python-poetry.org | python3 -",
        "delta": null,
        "end": null,
        "failed": false,
        "msg": "Did not run command since '/usr/local/bin/poetry' exists",
        "rc": 0,
        "start": null,
        "stderr": "",
        "stderr_lines": [],
        "stdout": "skipped, since /usr/local/bin/poetry exists",
        "stdout_lines": [
            "skipped, since /usr/local/bin/poetry exists"
        ]
    }
}
```

Reply:
Saves attempting to re-install poetry and gives the task a level of idempotency.